include/oneapi/tbb/task_group.h

```diff
@@ -187,6 +187,23 @@ class task_group_context : no_copy {
     };
     task_group_context_version my_version;
 
+  #if defined __APPLE__ && defined __ppc__
+    /** bool is 32 bits on Darwin ppc32 (but not on ppc64): https://gcc.gnu.org/bugzilla/show_bug.cgi?id=107590
+    https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/LowLevelABI/100-32-bit_PowerPC_Function_Calling_Conventions/32bitPowerPC.html. */
+    //! The context traits.
+    struct context_traits {
+        bool fp_settings        : 4;
+        bool concurrent_wait    : 4;
+        bool bound              : 4;
+        bool reserved1          : 4;
+        bool reserved2          : 4;
+        bool reserved3          : 4;
+        bool reserved4          : 4;
+        bool reserved5          : 4;
+    } my_traits;
+
+    static_assert(sizeof(context_traits) == 4, "Traits shall fit into 4 bytes.");
```

Comment:
I believe whole idea was to pack the traits into one byte. @pavelkumbrasev, please correct me if I'm wrong

Reply:
IMHO size of `bool` on Darwin is not related here The code asks compiler to use precisely 1 bit per field of the `context_traits` structure. 

As well gcc 4.8.5 for power can [compile](https://gcc.godbolt.org/z/1do8v9rh1) this

Reply:
@anton-potapov I readily admit I can be wrong regarding a cause, but as a matter of fact existing code fails on macOS PPC:
```
In file included from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_onetbb/onetbb/work/oneTBB-2021.7.0/src/tbb/scheduler_common.h:24,
                 from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_onetbb/onetbb/work/oneTBB-2021.7.0/src/tbb/task_dispatcher.h:24,
                 from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_onetbb/onetbb/work/oneTBB-2021.7.0/src/tbb/arena.cpp:17:
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_onetbb/onetbb/work/oneTBB-2021.7.0/src/tbb/../../include/oneapi/tbb/task_group.h:202:42: error: static assertion failed: Traits shall fit into one byte.
  202 |     static_assert(sizeof(context_traits) == 1, "Traits shall fit into one byte.");
      |                   ~~~~~~~~~~~~~~~~~~~~~~~^~~~
compilation terminated due to -Wfatal-errors.
make[2]: *** [src/tbb/CMakeFiles/tbb.dir/arena.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
In file included from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_onetbb/onetbb/work/oneTBB-2021.7.0/src/tbb/scheduler_common.h:24,
                 from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_onetbb/onetbb/work/oneTBB-2021.7.0/src/tbb/mailbox.h:23,
                 from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_onetbb/onetbb/work/oneTBB-2021.7.0/src/tbb/arena_slot.h:28,
                 from /opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_onetbb/onetbb/work/oneTBB-2021.7.0/src/tbb/arena_slot.cpp:17:
/opt/local/var/macports/build/_opt_PPCSnowLeopardPorts_devel_onetbb/onetbb/work/oneTBB-2021.7.0/src/tbb/../../include/oneapi/tbb/task_group.h:202:42: error: static assertion failed: Traits shall fit into one byte.
  202 |     static_assert(sizeof(context_traits) == 1, "Traits shall fit into one byte.");
      |                   ~~~~~~~~~~~~~~~~~~~~~~~^~~~
compilation terminated due to -Wfatal-errors.
```
[onetbb_static_assert_1.log](https://github.com/oneapi-src/oneTBB/files/10095359/onetbb_static_assert_1.log)

Reply:
@anton-potapov If it is preferable, we can alternatively not perform that static assert check on macOS ppc32.

Reply:
@barracuda156 As an alternative, you might look into GCC's `-mone-byte-bool` flag. I used this flag to reconcile bool-size issues with `mozjs`. However, if this is a public header, then client projects will also need that flag passed. (With mozjs I modified the pkg-config file to include the flag.)

https://gcc.gnu.org/onlinedocs/gcc/Darwin-Options.html

Reply:
@evanmiller Did not know about that, interesting. I cannot judge however if it is desirable in this case or not.

@iains @anton-potapov @glaubitz @pavelkumbrasev @reble What do you think?

P. S. It certainly does not hurt to try building with this flag and running tests. I will update on that.

Reply:
@anton-potapov @glaubitz @pavelkumbrasev @reble @evanmiller So passing `-mone-byte-bool` as cxxflag indeed removed the need of patching `struct context_traits` and `sizeof(task_group_context)`. Without the flag correct values are 4 and 136, respectively.

Obviously passing `-mone-byte-bool` flag simplifies changes, however I do not know if it may cause any adverse effects.

@iains Since this is both Darwin- and GCC-related matter, could you please give an advice? Without looking into specifics (I am aware you are very busy atm), but generally speaking for the flag itself.

Reply:
UPD. Hmm, probably not a great choice: while I cannot run the whole testsuite in Rosetta (being away from my PowerMacs), since it freezes, the very beginning is much worse when the build is done with `-mone-byte-bool`:
```
/opt/local/bin/ctest --force-new-ctest-process 
Test project /opt/local/var/macports/build/_opt_PPCRosettaPorts_devel_onetbb/onetbb/work/build
        Start   1: test_tick_count
  1/136 Test   #1: test_tick_count ..........................***Failed    0.07 sec
        Start   2: test_allocators
  2/136 Test   #2: test_allocators ..........................   Passed    0.05 sec
        Start   3: test_arena_priorities
  3/136 Test   #3: test_arena_priorities ....................Bus error***Exception:   0.16 sec
        Start   4: test_dynamic_link
  4/136 Test   #4: test_dynamic_link ........................   Passed    0.04 sec
        Start   5: test_collaborative_call_once
  5/136 Test   #5: test_collaborative_call_once .............***Failed   70.86 sec
        Start   6: test_concurrent_lru_cache
  6/136 Test   #6: test_concurrent_lru_cache ................   Passed    0.04 sec
        Start   7: test_concurrent_unordered_map
  7/136 Test   #7: test_concurrent_unordered_map ............Subprocess aborted***Exception:   4.21 sec
```